### PR TITLE
api: refactor to use early-return pattern

### PR DESCRIPTION
**Pull request by leonardojoao**

### CODE REVIEW

File: api/deviceadm/service.go [hunk 1]
```diff
@@ -88,24 +88,30 @@ func (s *service) DeleteDevice(ctx context.Context, uid models.UID, tenant, user
...
-			device.Name = name
-			if err := validate.Struct(device); err == nil {
-				otherDevice, _ := s.store.DeviceGetByName(ctx, name, tenant)
-				if otherDevice == nil {
-					return s.store.DeviceRename(ctx, uid, name)
-				}
-			}
-		}
+	if device.Name == name {
+		return			// I need return the "Error name is equal"
```
Comment (otavio) [inline api/deviceadm/service.go:102]: return nil // No operation needed.

File: api/deviceadm/service.go [hunk 2]
```diff
@@ -88,24 +88,30 @@ func (s *service) DeleteDevice(ctx context.Context, uid models.UID, tenant, user
...
 	}
-	return ErrUnauthorized
+
+	otherDevice, err := s.store.DeviceGetByName(ctx, name, tenant)
+	if err != nil {
+		return err		// Not Found
+	}
+
+	if otherDevice != nil {
+		return 			// You already have this device name in the namespace
```
Comment (otavio) [inline api/deviceadm/service.go:111]: return ErrDuplicatedDeviceName

**Review by otavio — CHANGES_REQUESTED** [review]
The commit log could be reworded as:

api: refactor to use early-return pattern

File: api/deviceadm/service.go (hunk 1 above)
Comment (leonardojoao) [reply api/deviceadm/service.go:102]: Could i return a specific error stating that the name is the same?

Comment (otavio) [reply api/deviceadm/service.go:102]: I think it does make sense. And who uses it, need to handle this as well.

File: api/deviceadm/service.go [hunk 3]
```diff
@@ -88,24 +88,30 @@ func (s *service) DeleteDevice(ctx context.Context, uid models.UID, tenant, user
 func (s *service) RenameDevice(ctx context.Context, uid models.UID, name, tenant, username string) error {
 	err := s.isNamespaceOnwer(ctx, tenant, username)
 	if err != nil {
-		return err
+		return ErrUnauthorized
+	}
+
+	device, err := s.store.DeviceGetByUID(ctx, uid, tenant)
+	if err != nil {
+		return err		// Not Found
```
Comment (leonardojoao) [inline api/deviceadm/service.go:96]: I could return ErrDeviceNotFound?

File: api/deviceadm/service.go (hunk 1 above)
Comment (leonardojoao) [reply api/deviceadm/service.go:102]: Yes, I will do this

**Review by otavio — DISMISSED** [review]

**Review by gustavosbarreto — APPROVED** [review]

**Pull request merged**